src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/ExportIPOs/ExportIpoRepository.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Transactions;
+using Dapper;
+using Equinor.ProCoSys.Common.Misc;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using Microsoft.EntityFrameworkCore;
+
+
+namespace Equinor.ProCoSys.IPO.Infrastructure.Repositories.ExportIPOs;
+
+
+public class ExportIpoRepository : DapperRepositoryBase, IExportIpoRepository
+{
+    public ExportIpoRepository(IPOContext context) : base(context) { }
+
+    public async Task<List<Invitation>> GetInvitationsWithIncludesAsync(
+        List<int> invitationIds,
+        IPlantProvider plantProvider,
+        CancellationToken cancellationToken)
+    {
+        await using (var connection = Context.Database.GetDbConnection())
+        {
+            await connection.OpenAsync(cancellationToken);
+            await using (var transaction = await connection.BeginTransactionAsync())
+            {
+                transaction.Commit();
+                var parameters = new { plant = plantProvider.Plant };
+                using (var results = await connection.QueryMultipleAsync("GetInvitations", parameters, commandType: CommandType.StoredProcedure))
+                {
+                    var invitations = await results.ReadAsync<Invitation>();
```

Comment:
Either `var invitations = (await results.ReadAsync<Invitation>()).ToList();`
or create a new var invitationList = invitations.ToList();

simplify the return invitations.ToList() on line 72 to either return invitations; or return invitationList; accordantly

Reply:
No action taken since last PR. Still possible multiple enumeration waring on invitations variable:
![image](https://github.com/equinor/procosys-invitation-for-punch-out-api/assets/19545528/86958182-0aac-4083-b614-f16441a32224)